### PR TITLE
Add Support for DataType, ValueType, Primitive, and Enumeration

### PR DESCRIPTION
**Pull request by danyeaw**
This PR adds initial support for UML DataType, Primitive, and Enumeration, as well as SysML ValueType.

### PR Checklist
Please check if your PR fulfills the following requirements:

- [X] I have read, and am following the [Contributor guide](https://github.com/gaphor/gaphor/blob/master/CONTRIBUTING.md)
- [X] I have read and I understood the [Code of Conduct](https://github.com/gaphor/gaphor/blob/master/CODE_OF_CONDUCT.md)

### PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes

### What is the current behavior?
It is not possible to create these elements in a diagram. 

Issue Number: #701 

### What is the new behavior?
We now support them :tada: 

### Does this PR introduce a breaking change?
- [ ] Yes
- [X] No

### CODE REVIEW

**Comment by danyeaw** [comment]
@amolenaar I would like the enumeration literals to show nested under the enumeration in the Namespace. Where do we set which types of elements are to be displayed as rows in the Namespace? Does setting the owner result in being picked up when the `namespacingmodel.py` module is iterating?

**Comment by amolenaar** [comment]
@danyeaw The Namespace view uses the owner-ownedElement relation. So if one is owned by the other it should show up.

**Comment by danyeaw** [comment]
@amolenaar I think this is ready for a review. I noticed a few things that could probably have their own PRs:
1. Attributes added to a class don't update unless you cycle the Show Attributes slider
2. Sometimes Enumerations can get stuck on the screen, and can't be moved up any further. If I add another element to the diagram then they start working again
3. ValueTypes should be able to have their own ValueTypes (nested) in a separate compartment.

**Review by amolenaar — APPROVED** [review]
Nice work. Sorry the review took so long.

File: gaphor/C4Model/toolbox.py [hunk 1]
```diff
@@ -65,7 +65,7 @@ def component_config(new_item):
             "c4-software-system",
             gettext("Software System"),
             "gaphor-c4-software-system-symbolic",
-            "s",
+            "<Shift>S",
```
[the remark this answers is not in the document]
Comment (danyeaw) [reply gaphor/C4Model/toolbox.py:68]: I 100‰ agree, unfortunately with adding SysML, C4, and RAAML we are completely out of shortcuts. Could we make some Ctrl+Alt or some other modifier combination?

**Pull request merged**